Provide element template `keywords` to popup-menu search

Buckwich opened the pull request:
Related to https://github.com/camunda/camunda-modeler/issues/4853

Depends on https://github.com/bpmn-io/diagram-js/pull/970

### Proposed Changes

To make use of the new keyword property it has to be forwarded to the popup menu as an entry. This PR just adds that property to the according provider.

<!--

Add relevant context (issue fixed or related to), 
a capture of the UI changes (if any) as well as 
steps to try out your changes.

--> 

### Try out

`npm run start`

try to create, append, or replace an element. you should be able to find templates also via their keyword. for example "another keyword" for the keyword template

<img width="369" alt="image" src="https://github.com/user-attachments/assets/58d1293f-cef7-4518-af82-7181eeea04ce" />

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [ ] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

<!--

Thanks for creating this pull request! ❤️

-->

Buckwich committed:
feat: make keywords searchable

Related to https://github.com/camunda/camunda-modeler/issues/4853

nikku commented on lib/element-templates/create-menu/ElementTemplatesCreateProvider.js:
```diff
@@ -80,6 +81,7 @@ ElementTemplatesCreateProvider.prototype.getTemplateEntries = function() {
       documentationRef: template.documentationRef,
       imageUrl: icon.contents,
       group: category || defaultGroup,
+      keywords,
```
Do we invent a separate `keywords` entry here? Why not re-use the existing `search` mechanism that the popup menu already recognizes?

Buckwich replied:
i was concerned about introducing a breaking change. Eg someone is still using an old version of diagram-js that only can handle search:string, but for some reason loads a template with search:string[]

nikku replied:
The upgrade will be as simple as relying on the new minor version. We typically accept that risk, when the alternative is to blow up the core editor scope unnecessarily.

nikku replied:
CC @barmac

nikku replied:
What we could do is to add a `diagram-js >= XXX` peerDependency range, but then again, we assume our libraries are updated _as a block_ and not individually. An unacceptable break would be a bpmn-js minor (or patch) update that breaks with dependencies.

barmac replied:
Thanks Nico for tagging. Breaking change could only be introduced in an upstream project, e.g. in case that the menu starts to accept only `string[]`, and fails to read `string`. This project could declare with which version of bpmn-js it works, and pass new value type.

Buckwich replied:
i'll drop the commit then. is it expected that element templates could also contain search? cf. [b720111](https://github.com/bpmn-io/bpmn-js-create-append-anything/pull/50/commits/b7201114ca0a253d9598fae1df157acdf18c6cd6)

nikku replied:
> is it expected that element templates could also contain search?

No, they contain `keywords`, `description`, and `name`.

Buckwich replied:
i've removed support for search

nikku reviewed (changes requested):
I think we should not invent another mechanism to pass stuff to the core diagram-js popup-menu. When the menu already supports _search=string_, then it could also easily support _search=string[]_.

nikku reviewed (commented):
Good stuff. Let's keep this as a draft, and bump bpmn-js with it (supporting `search=string[]`).

nikku commented:
Waiting for string[] support for search in https://github.com/bpmn-io/diagram-js/pull/966

Buckwich commented on lib/element-templates/create-menu/ElementTemplatesCreateProvider.js:
```diff
@@ -80,13 +81,16 @@ ElementTemplatesCreateProvider.prototype.getTemplateEntries = function() {
     };
   });
 
+  console.log(templateEntries);
```
todo

Buckwich committed:
chore: update to @bpmn-io/element-template-chooser@2.0.0

chore: make @bpmn-io/element-template-chooser a DEV dependency

  Not actually used in this library.

nikku reviewed (approved):
Works!

Buckwich commented:
waiting for v2.0.0 of @bpmn-io/element-template-chooser to properly fix

Buckwich committed:
chore(CHANGELOG): update to v1.0.0

nikku commented on CHANGELOG.md:
```diff
@@ -6,6 +6,15 @@ All notable changes to [bpmn-js-create-append-anything](https://github.com/bpmn-
+## 1.0.0
+
+* `FEAT`: forward keywords to make them searchable ([#50](https://github.com/bpmn-io/bpmn-js-create-append-anything/pull/50))
+* `DEPS`: update bpmn dependencies ([#50](https://github.com/bpmn-io/bpmn-js-create-append-anything/pull/50))
```
Be clear what exactly you update, and what new peer dependencies (if declared) are now required.

Buckwich replied:
done, but diagram-js is only a transitive dependency, i still put it in separately as a breaking change

nikku replied:
> done, but diagram-js is only a transitive dependency, i still put it in separately as a breaking change

In fact we rely on diagram-js >= 15.3.0 in the way we now source `search=string[]`, hence the requirement is not of transitive nature.

Buckwich replied:
right, I think i mixed up two things. now with diagram js as a peer this is no longer true

nikku commented on CHANGELOG.md:
```diff
@@ -6,6 +6,15 @@ All notable changes to [bpmn-js-create-append-anything](https://github.com/bpmn-
+## 1.0.0
+
+* `FEAT`: forward keywords to make them searchable ([#50](https://github.com/bpmn-io/bpmn-js-create-append-anything/pull/50))
+* `DEPS`: update bpmn dependencies ([#50](https://github.com/bpmn-io/bpmn-js-create-append-anything/pull/50)): `bpmn-js@^18.6.0`, `bpmn-js-properties-panel@^5.35.0`, `bpmn-js-element-templates@^2.5.3`, `@bpmn-io/element-template-chooser@^2.0.0`
```
Which of these bumps are actually required? I think none of them. And if we now have a "minimum diagram-js requiremeent", let's add it to `peerDeependecies".

nikku replied:
Even better, let's anchor at `bpmn-js` for peerDepenedncies. The minimum required version would be `bpmn-js@18.6.0` then.

Buckwich replied:
@bpmn-io/element-template-chooser@^2.0.0 is the required one because thats the transitive one for diagram js

Buckwich replied:
or bpmn-js that also includes diagram-js but not as peer

Buckwich replied:
> The minimum required version would be bpmn-js@18.6.0 then.

I don't think so, even bpmn-js@18.4.0 should work as long as diagram-js is >= 15.3.0

we should move bpmn-js out of devDependencies to normal dependencies.

nikku replied:
Cf. https://github.com/bpmn-io/bpmn-js-create-append-anything/pull/50#pullrequestreview-2792137265 how I solved this. Still a breaking change due to `diagram-js >= 15.3.0` requirement, but the chooser is not an actual dependency.

nikku commented on package.json:
```diff
@@ -41,14 +41,20 @@
   "module": "dist/index.es.js",
+  "dependencies": {
+    "@bpmn-io/element-template-chooser": "^2.0.0",
+    "bpmn-js": "^18.6.0"
```
We usually don't do this, as this library "builds on top of bpmn-js" anyway. If a certain bpmn-js version is needed, then we `peerDepend` on it.

Yes, we could rely on de-duplication of bundlers, but for most parts (cf. our other projects) we simply rely on bpmn-js being there, when you pull in an extension for it.

nikku committed:
deps!: peer depend on diagram-js >= 15.3.0

We now rely on `search=string[]` support, shipped with diagram-js core.

nikku reviewed (approved):
Slight adjustments to changelog and update:

* Removed `bpmn-js` dependency (again) as per previous comment
* Removed `@bpmn-io/element-template-chooser` dependency, it is actually a `DEV` only dependency
* Still we rely on `diagram-js >= 15.3.0`, as we now provide `search=string[]` :arrow_right: peer dependency is needed

Updated changelog accordingly.

nikku commented:
Released as [`v1.0.0`](https://github.com/bpmn-io/bpmn-js-create-append-anything/releases/tag/v1.0.0)
